### PR TITLE
Second attempt to enable custom tree for 'zinit ls' command

### PR DESCRIPTION
**Pull request by nbrown**
Enable users to specify the zinit ls command via ZINIT[LIST_COMMAND] and
provide default commands for exa and tree. Prefer exa if it is installed.

Addresses issue https://github.com/zdharma-continuum/zinit/issues/170

I cherry-picked the original changes to a clean branch and then updated the docs to attempt to address the documentation action issues.

### CODE REVIEW

**Comment by vladdoster** [comment]
oof

**Comment by alichtman** [comment]
Whatever. Let the test fail.

**Review by alichtman — APPROVED** [review]
Haven't tested it, but looks good to me

**Review by alichtman — APPROVED** [review]
Haven't tested it, but looks good to me

**Comment by nbrown** [comment]
Well, I managed to make it work by using a docker container used in https://github.com/nektos/act

My conclusion is that there is something different between my macOS en_US.UTF-8 and the one used in the github actions containers.

**Comment by alichtman** [comment]
This PR shouldn't contain a massive docs update. That should already be on master. I don't have time to jump into debugging it but I won't merge it with an ~800 line addition diff

**Comment by vladdoster** [comment]
@nbrown great job on getting this to pass 

giphy

**Comment by vladdoster** [comment]
@nbrown Since you mentioned this, I did some digging and think I found the root issue. 

I went to our Dockerfile and saw we don't explicitly set the locale, which seemed odd. A few GitHub issues later and I found this issue:

https://github.com/dotnet/runtime/issues/962

Your hypothesis seems to be spot on and created an issue to track it.

**Comment by nbrown** [comment]
> This PR shouldn't contain a massive docs update. That should already be on master. I don't have time to jump into debugging it but I won't merge it with an ~800 line addition diff

Sorry for the large doc change, I don't know how you want to handle it but two choices are minimal doc changes that don't pass the documentation check or a large change that passes the check. I'll gladly revert it to not pass that check if that's what is preferred.

**Comment by alichtman** [comment]
Yeah, let's keep the PR tight and drop the docs change. We'll get the docs test sorted out soon (maybe, idk -- no promises) but I'm fine with it failing here.

**Comment by nbrown** [comment]
> Yeah, let's keep the PR tight and drop the docs change. We'll get the docs test sorted out soon (maybe, idk -- no promises) but I'm fine with it failing here.

Done. I reverted the doc fix so it's now a minimal change.

On the docs test, I'll leave some info on that bug. I think it primarily comes down to a sorting (collation) inconsistency between Linux and macOS/BSD based on my reading and testing.

File: zinit-autoload.zsh [hunk 1]
```diff
@@ -3246,7 +3246,8 @@ EOF
         setopt localoptions extendedglob nokshglob noksharrays
         builtin cd -q "${ZINIT[SNIPPETS_DIR]}"
         local -a list
-        list=( "${(f@)"$(LANG=en_US.utf-8 tree -L 3 --charset utf-8)"}" )
+        local -x LANG=en_US.utf-8
```
Comment (alichtman) [inline zinit-autoload.zsh:3249]: Is this required by `exa`?

Comment (nbrown) [reply zinit-autoload.zsh:3249]: Actually no. I took the issue's suggestion at face value, but as far as I can tell from exa's source code, it doesn't rely on locale for sorting.

**Review by vladdoster — COMMENTED** [review]
This check evaluates to true even though I have exa, but no tree.

File: zinit-autoload.zsh (hunk 1 above)
Comment (vladdoster) [inline zinit-autoload.zsh:3249]: The [prior check](https://github.com/zdharma-continuum/zinit/pull/190/files#diff-6a3752f1e40118bbfc5b0e05b9fa86aaffbd897b4c36f760bb3221a862f68343R3240-R3244) evaluates to false even though I have exa, but no tree.

**Review by vladdoster — CHANGES_REQUESTED** [review]

**Review by vladdoster — APPROVED** [review]

**Comment by vladdoster** [comment]
Closing in favor of #221 due to lack of update on the PR.

**Pull request closed without merging**